bam_sort.c

```diff
@@ -159,6 +160,11 @@ typedef enum {Coordinate, QueryName, TagCoordinate, TagQueryName, MinHash, Templ
 static SamOrder g_sam_order = Coordinate;
 static char g_sort_tag[2] = {0,0};
 
+#ifdef isdigit
+#    undef isdigit
+#endif
+#define isdigit(c) ((c)<='9' && (c)>='0')
```

Comment:
Overriding a libc function like this looks dodgy.  I think we should call the macro something like `is_digit`, replace `isdigit` with `is_digit` in `strnum_cmp()`, and remove `#include <ctype.h>` because it's no longer needed.

Reply:
Fair comment, although looking at the entire function I see it's unnecessarily convoluted anyway.

I have a rewrite which is considerably quicker in gcc, and no worse in clang which was already better than gcc (entire sort is 20% less CPU, including all the BAM/SAM parsing and I/O).